core/escrow.js

```diff
@@ -2,95 +2,144 @@
+    }
+
+    // Promise wiring
+
+    const moneyPayment = makePromise();
+    const moneyTransfer = makeTransfer(moneyNeeded, moneyPayment.p);
+
+    const stockPayment = makePromise();
+    const stockTransfer = makeTransfer(stockNeeded, stockPayment.p);
+
+    // TODO Use cancellation tokens instead.
+    const aliceCancel = makePromise();
+    const bobCancel = makePromise();
+
+    // Set it all in motion optimistically.
+
+    const decisionP = Promise.race([
+      Promise.all([moneyTransfer.phase1(), stockTransfer.phase1()]),
+      aliceCancel.p,
+      bobCancel.p,
+    ]);
+    decisionP.then(
+      _ => {
+        moneyTransfer.phase2();
+        stockTransfer.phase2();
       },
-      abort(reason) {
-        winnings.reject(reason);
-        refund.res(escrowP);
-      },
-      getWinnings() {
-        return winnings.p;
-      },
-      getRefund() {
-        return refund.p;
+      reason => {
+        moneyTransfer.abort(reason);
+        stockTransfer.abort(reason);
       },
+    );
+
+    // Seats
+
+    const aliceSeat = harden({
+      offer: moneyPayment.res,
+      cancel: aliceCancel.reject,
+      getWinnings: stockTransfer.getWinnings,
+      getRefund: moneyTransfer.getRefund,
+    });
+
+    const bobSeat = harden({
+      offer: stockPayment.res,
+      cancel: bobCancel.reject,
+      getWinnings: moneyTransfer.getWinnings,
+      getRefund: stockTransfer.getRefund,
+    });
+
+    return harden({
+      left: inviteMaker.make('left', aliceSeat),
+      right: inviteMaker.make('right', bobSeat),
     });
-  }
-
-  // Promise wiring
-
-  const moneyPayment = makePromise();
-  const moneyTransfer = makeTransfer(moneyNeeded, moneyPayment.p);
-
-  const stockPayment = makePromise();
-  const stockTransfer = makeTransfer(stockNeeded, stockPayment.p);
-
-  // TODO Use cancellation tokens instead.
-  const aliceCancel = makePromise();
-  const bobCancel = makePromise();
-
-  // Set it all in motion optimistically.
-
-  const decisionP = Promise.race([
-    Promise.all([moneyTransfer.phase1(), stockTransfer.phase1()]),
-    aliceCancel.p,
-    bobCancel.p,
-  ]);
-  decisionP.then(
-    _ => {
-      moneyTransfer.phase2();
-      stockTransfer.phase2();
-    },
-    reason => {
-      moneyTransfer.abort(reason);
-      stockTransfer.abort(reason);
-    },
-  );
-
-  // Seats
-
-  const aliceSeat = harden({
-    offer: moneyPayment.res,
-    cancel: aliceCancel.reject,
-    getWinnings: stockTransfer.getWinnings,
-    getRefund: moneyTransfer.getRefund,
-  });
-
-  const bobSeat = harden({
-    offer: stockPayment.res,
-    cancel: bobCancel.reject,
-    getWinnings: moneyTransfer.getWinnings,
-    getRefund: stockTransfer.getRefund,
-  });
-
-  return harden({
-    left: inviteMaker.make('left', aliceSeat),
-    right: inviteMaker.make('right', bobSeat),
-  });
-}
-
-const escrowExchangeSrc = `(${escrowExchange})`;
-
-export { escrowExchange, escrowExchangeSrc };
+  },
+
+  checkAmount: (amount, payment) => {
```

Comment:
These parameter names are terrible. I suggest `(allegedInviteAmount, expectedTerms)`. There are many amounts running around, and invite amounts are not the first one thinks of. Here, the amount is what the caller is checking against their expectations. And this amount is obtained from an untrusted object as well.

The `expectedTerms` is how the caller expresses what they expect. Note that the caller expesses what excrowExhange installation they expect according to the receiver of the check message. However, the check method must check the alleged installation against this expectation as well, or the assurances from the rest of the checking is meaningless.

Even when checking the full terms, rather than partial checking, the caller still needs to check that the seat they'd be getting is the seat they expect. They need to provide this as an additional parameter. This would also solve the ambiguity problem with the partial checking below.